BSB_LAN/include/mqtt_handler.h

```diff
@@ -200,8 +203,20 @@ boolean mqtt_connect() {
       default: bufsize = 32; break;
     }
     MQTTPubSubClient->setBufferSize(bufsize);
+    mqtt_reconnect_timer = 0;
+    first_connect = true;
   }
   if (!MQTTPubSubClient->connected()) {
+    if (!first_connect && !mqtt_reconnect_timer) {
+      // We just lost connection, don't try to reconnect immediately
+      mqtt_reconnect_timer = millis();
+      printlnToDebug(PSTR("MQTT connection lost"));
+      return false;
+    }
+    if (mqtt_reconnect_timer and millis() - mqtt_reconnect_timer < 1000)
```

Comment:
Please make sure to use `&&` instead of `and` to keep the code consistent. Also, no indentions without curly brackets.

Reply:
Also: Why is the test for mqtt_reconnect_timer being true/1 needed here? The rest of the expression will evaluate to false even if mqtt_reconnect_timer is 0 because it will be more than 1000 once it has reached this point in loop().

Reply:
No, I get the reason behind that, and also that `mqtt_reconnect_timer` needs to be 0 to prevent immediate reconnection. I was wondering if your
`if (mqtt_reconnect_timer and millis() - mqtt_reconnect_timer < 1000)`
has the same effect as 
`if (millis() - mqtt_reconnect_timer < 1000)`
because if `mqtt_reconnect_timer` is zero, the if clause is still not executed because millis() - 0 will always be greater than 1000 by the time this line is executed. Or did I overlook something here?

Reply:
As for Asyc-Mqtt-Client, I agree this would be more useful, but don't forget that BSB-LAN aims to work on both Arduino Due and ESP32. Unless making use of different libraries is as simple as compiling different libraries, the benefits usually do not outweigh the extra effort it takes to maintain a code base that has already traversed across three different platforms (Arduino Mega, Due, ESP32).